### DISCUSSION TITLE
Create level by concatenating existing levels

### DISCUSSION THREAD

**Discussion by kascp** [atoti/atoti#814] [Q&A] [answered]
Hi

I am able to create a structure in my levels as shown in code below. Here, I create new hierarchy from Content Name and Country Name so that the columns are still shown separately.
What I want to do: I want to concatenate both string values into a new level in the cube, i.e. target would be a new level REGION_FINAL = continentDimension["CONTINENT_NAME"] + ' / ' + countryDimension["COUNTRY_NAME"]

E.g. the level would directly hold value "Europe / Germany"

Obviously I could do it in pandas and have it directly as a separate dimension regionDimension but I dont think this is elegant. 

Do you have an idea to solve my problem?

Thanks
Philipp

h.update(
    {
        ("HELP DIMENSIONS", "REGION STRUCTURE"): [continentDimension["CONTINENT_NAME"], countryDimension["COUNTRY_NAME"]],
    }
)

**Comment by tibdex** [comment, marked as answer]
Hi Philipp, that's not supported. What's wrong with having the continent and country be their own level?

**Comment by kascp** [comment]
Hi
Thanks, then I will create additional dimension for it.
For me it is useful because I want to show 3 levels at the same time and in a more compressed way (instead of columns next to each other). In my case, some of those 3 levels are often null, so concatenating them into one level makes it much thinner in the UI.
Thanks
Philipp

**Reply by tibdex** [reply]
Have you tried switching from a **Pivot table** to a **Tree table**?

![Tree table](https://github.com/atoti/atoti/assets/6574550/fa3eb5bf-5c98-4ba6-b3ef-a66068c70563)
